### DISCUSSION TITLE
Acknowledgement of Oven* software in scientific papers

### DISCUSSION THREAD

**Discussion by gehirndienst** [OvenMediaLabs/OvenMediaEngine#1395] [Help] [answered]
Hello,

Preamble: I don't know how to classify my question, sorry if "Discover" is not the appropriate choice :)

I am a PhD student from Germany and I want to use your software in my scientific research about controlling the publishing rate of RTP streams using Deep Reinforcement Learning. I find OME to be perfect for me because it is open-source, has clean and concise C++ code for such a big software project, and provides exactly what I need: an intermediate ingester where one can play with changing the quality preset (rendition) of RTSP video streams (IP-cameras) in uplink scenarios and deploying it within a docker for a real testbed. If you wonder what are exact changes planned, so far it is everything that goes in the scope of ABR for WebRTC (further after the highlighted line below). I plan to replace the current ABR with our IPC-bridged DRL model implemented in Python and pre-trained in WLAN/5G simulation so that there is going to be a new controller class(es) integrated into OME for the bitrate control. 

https://github.com/AirenSoft/OvenMediaEngine/blob/d9316c73f9d8a763d3f9ebdba396225815d24180/src/projects/publishers/webrtc/rtc_session.cpp#L659
 
My question is, how should we acknowledge, i.e., reference your software? It is under GPLv3, so of course we will publish our modified version under the same license in a public repo as well. The standard practice is also to reference the "reference paper" in your paper, if available, otherwise the website. This reference paper allows tagging the main developers hence the authors of this paper personally. Do you have such a paper or have you had already experience and policies for referencing/citing your software?

Best
Nikita

**Comment by getroot** [comment, marked as answer]
Hello,

There is no specific citation or referencing policy for OvenMediaEngine. If you can include the GitHub URL in your paper in any way, it would be helpful to many others as well: https://github.com/AirenSoft/OvenMediaEngine

Also, one correction to note is that OvenMediaEngine is licensed under AGPLv3, not GPLv3. It would be a good idea to fork and make modifications on GitHub. If you believe that your research findings could be valuable to many, please consider submitting a pull request (PR) to us. I will review it and merge it into the OvenMediaEngine project.

Please keep me informed when your paper is published. I am also interested in your research.

Thank you.
